### PR TITLE
Add 2.4 Changelog

### PR DESCRIPTION
**Pull request by jakirkham**
Includes some other Changelog entries for 2.4.

xref: https://github.com/zarr-developers/zarr-python/issues/529

cc @jrbourbeau

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [x] Changes documented in docs/release.rst
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

File: docs/release.rst [hunk 1]
```diff
@@ -65,6 +65,13 @@ Upcoming Release
 Bug fixes
 ~~~~~~~~~
 
+* Fix Sqlite Store Wrong Modification.
+  By :user:`potter420 <potter420>`; :issue:`440`.
```
Comment (jakirkham) [inline docs/release.rst]: @potter420, are you ok being listed here? How would you like to be referred to?

**Comment by jakirkham** [comment]
@jrbourbeau, could you please take a look and let me know if I'm missing anything?

**Comment by jrbourbeau** [comment]
Thanks @jakirkham, apologies for not getting to this earlier. I'm stepping away from my laptop for a few hours, but will review later this evening

File: docs/release.rst (hunk 1 above)
Comment (potter420) [reply docs/release.rst]: I would like to be referred to as Tommy Tran. 
My current name is kinda goofy.

Comment (jakirkham) [reply docs/release.rst]: Sounds good. Thanks. Have pushed in the latest commit. Can you please let me know whether I got that correct?

**Comment by jakirkham** [comment]
Thanks for doing that @jrbourbeau! Looks good. Added those missing pieces and merged with `master`. Please let me know if anything else is needed 🙂

**Comment by jrbourbeau** [comment]
I'll go through the release procedure docs and see if there are any updates to be made there. Other than that, we should be good to release (woo!)

**Comment by jakirkham** [comment]
Push away! 😄

**Pull request merged**